### PR TITLE
feat(trends): Compact series math selector

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

The options of two math selector's sections, "Event aggregation" and "Property aggregation", didn't mesh well together logically. The issue is, the latter section was just a collection of 8 math functions, that in reality were just 8 variations of one aggregation mode ("Property value").

This meant the list was very long. But now an additional problem appeared with the introduction of "Count per user" aggregation (#12173): That _also_ is an average/minimum/etc., but it _isn't_ "Property aggregation", and the list would be completely unwieldy if all event count per user variants were their own select option. That meant I had to put the new mode into its own taxonomic selector tab, but that hurts discoverability a lot – it's not where one would expect this mode at all.

## Changes

This PR reworks the "Property aggregation" section into a "Property value [function]" option with a nested select. This way count per user is "Count per user [function]".

| Before | After |
| --- | --- |
| before | after |

"Count per user" breaking change: to accommodate this select pattern more easily, event count per user is now specified differently in filters.
Before, `math` was `avg`/`min`/etc. – same as property value aggregation – the difference was just that instead of `math_property` being some event property key, it was a special `"__event_count_per_actor"` value. This was elegant in a way and meshed well with "Event count per user" being a taxonomic selector option, but maintaining this with the new selector structure was really messy, since `MathSelector` is not made to update `math_property` – there's a separate conditional selector for that.

I anticipated I might go this way when implementing #12173, so I put that behind a feature flag… But I just realized I immediately rolled the feature out to everyone on Cloud instead of just our team. 🙈 So on US Cloud there are now 5 insights from other users, and those need to be migrated. This is still very few though (fortunately the taxonomic selector really didn't expose this well), so I'll probably just run a manual UPDATE query for those insights.

## How did you test this code?

Updated the relevant E2E test. Also lots of manual clicking around.

### CODE REVIEW

**Review by mariusandra — APPROVED** [review]
1. Likely not a new bug here, but when running a "count per user" with a breakdown, the query fails:

```sql
SELECT groupArray(value)
  FROM (
        SELECT replaceRegexpAll(JSONExtractRaw(properties, '$geoip_city_name'), '^"|"$', '') AS value,
               avg(intermediate_count) as count
          FROM events e
         WHERE team_id = 1
           AND event = '$pageview'
           AND toDateTime(timestamp, 'UTC') >= toDateTime('2022-10-14 00:00:00', 'UTC')
           AND toDateTime(timestamp, 'UTC') <= toDateTime('2022-10-21 23:59:59', 'UTC')
         GROUP BY value
         ORDER BY count DESC, value DESC
         LIMIT 25
        OFFSET 0
       )
```

fails with the error:

```
DB::Exception: Missing columns: 'intermediate_count' while processing query:
```

2. Unlike your screenshot, I see this as the second item:
image
This is probably better 🤔 

3. Looks like a nice refactor
4. Looks like code
5. Great improvement overall

**Comment by lharries** [comment]
flyby feedback so might be out of scope: this definitely looks in the right direction. 

I imagine many of our users will also ask to apply these statistics to other entities apart from users - i.e. session/organization/project/instance. Note: I haven't heard this specifically requested from a user but I've wanted this myself for organization analysis

One way to do this might be to change this:
Screenshot Cococart Event Counts • Insights • PostHog (Google Chrome) 2022-10-21 at 09 18@2x


to the following dropdowns:
**event, entity/property, group/aggregation** (or a different ordering)

e.g. "user sign up", "organizations", "unique"

where:
event is what we currently have
entity/property = user, session, organization, instance, project, customer
group/aggregation = total, unique, weekly active, monthly active, average, min, 90th percentile etc.

currently I think we merge the aggregations and the entities dropdown into one with a more limited set of options

I briefly checked what some other companies do, it seems Company A does this but Company B does not - and Company A's product seems more powerful https://docs.google.com/document/d/1NzBIjl9xiIC-aPFhPoIcJNqbloXxE9PwY233MB_T2Ek/edit

**Comment by Twixes** [comment]
Yeah, forgot to update the screenshot here @mariusandra I think logically putting "Count per actor between" "Total count" and "Unique <actors>" makes sense.
Breakdown is on the TODO list!

@lharries Groups are on the TODO list too. Your suggestion makes sense, I'll probably do something like that. Thanks for the competitive research. :)

File: frontend/src/scenes/groups/GroupsIntroduction.tsx [hunk 1]
```diff
@@ -42,27 +42,45 @@ export function GroupsIntroduction({ access }: Props): JSX.Element {
     )
 }
 
-export function GroupIntroductionFooter(): JSX.Element {
+export function GroupIntroductionFooter({ needsUpgrade }: { needsUpgrade: boolean }): JSX.Element {
```
Comment (Twixes) [inline frontend/src/scenes/groups/GroupsIntroduction.tsx:45]: As per https://github.com/PostHog/posthog/pull/12380/files/b403b183f92297c7b5f5042823896c6157022869#r1001803148 @kappa90

**Pull request merged**